adding_to_the_RDynLib_library.qmd

```diff
@@ -55,47 +54,33 @@ table(msLevel(s))
 
 this function takes as input the XcmsExperiment object, then it extract the MS2
 spectra using "featureSpectra()" function, after that we extract from the
-initial object only the MS3 fragments that have a parent ion in the result of 
+initial object only the MS3 fragments that have a parent ion in the result of
 "featureSpectra()" function, and then we do the same process for MS4.
 
 ```{r}
 ftms_all_levels <- function(ftms) {
-  
+
   ms2 <- featureSpectra(ftms, msLevel = 2)
   cat("Number of MS2 spectra found:", length(ms2), "\n")
-  
-  combined_spectra_list <- list() 
-  
-  for (origin in unique(spectraData(ms2)$dataOrigin)) {
+
+  res <- lapply(unique(ms2$dataOrigin), function(origin) {
```

Comment:
Using `lapply` instead of the `for` loop - then we don't have to think/create the result `list` - that is automatically returned by `lapply()`.